From e8cb624603324f27155e0dfe479aabc5c3bd3dfb Mon Sep 17 00:00:00 2001
From: Judith Mendez <jm@ti.com>
Date: Wed, 21 Feb 2024 18:33:15 -0600
Subject: [PATCH 13/27] arm64: dts: ti: k3-am62p5-sk: Enable GPIOs in GPIO1
 bank

Enable remaining GPIOs in GPIO1 bank for k3-am62p5-sk-rpi-hdr-ehrpwm
overlay.

While we are at it, fix epwm1 since pinmux conflicts with
main_uart1.

Disable dss0 node since pinmux conflicts with GPIO0_38.

Remove MCASP0 node (which disabled MCASP0) since it is not enabled
by default in the board DTS file.

Signed-off-by: Judith Mendez <jm@ti.com>
---
 .../boot/dts/ti/k3-am62p5-sk-rpi-hdr-ehrpwm.dtso   | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62p5-sk-rpi-hdr-ehrpwm.dtso b/arch/arm64/boot/dts/ti/k3-am62p5-sk-rpi-hdr-ehrpwm.dtso
index 9f627276eb50..7a56444f8247 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p5-sk-rpi-hdr-ehrpwm.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am62p5-sk-rpi-hdr-ehrpwm.dtso
@@ -27,14 +27,20 @@
 
 	rpi_header_gpio1_pins_default: rpi-header-gpio1-pins-default {
 		pinctrl-single,pins = <
+			AM62PX_IOPAD(0x01a4, PIN_INPUT, 7) /* (F24) MCASP0_ACLKX.GPIO1_11 */
+			AM62PX_IOPAD(0x01bc, PIN_INPUT, 7) /* (B21) SPI0_CLK.GPIO1_17 */
+			AM62PX_IOPAD(0x01c0, PIN_INPUT, 7) /* (B20) SPI0_D0.GPIO1_18 */
+			AM62PX_IOPAD(0x01c4, PIN_INPUT, 7) /* (C21) SPI0_D1.GPIO1_19 */
 			AM62PX_IOPAD(0x01d0, PIN_INPUT, 7) /* (A23) UART0_CTSn.GPIO1_22 */
+			AM62PX_IOPAD(0x01d8, PIN_INPUT, 7) /* (B23) MCAN0_TX.GPIO1_24 */
+			AM62PX_IOPAD(0x01dc, PIN_INPUT, 7) /* (F20) MCAN0_RX.GPIO1_25 */
 		>;
 	};
 
 	rpi_header_ehrpwm0_pins_default: rpi-header-ehrpwm0-pins-default {
 		pinctrl-single,pins = <
-			AM62PX_IOPAD(0x01ac, PIN_OUTPUT, 6) /* (G23) MCASP0_AFSR.EHRPWM0_A */
-			AM62PX_IOPAD(0x01b0, PIN_OUTPUT, 6) /* (G20) MCASP0_ACLKR.EHRPWM0_B */
+			AM62PX_IOPAD(0x01b4, PIN_OUTPUT, 2) /* (D20) SPI0_CS0.EHRPWM0_A */
+			AM62PX_IOPAD(0x01b8, PIN_OUTPUT, 2) /* (E20) SPI0_CS1.EHRPWM0_B */
 		>;
 	};
 
@@ -76,7 +82,8 @@
 	};
 };
 
-&mcasp0 {
+&dss0 {
+	/* Conflict with GPIO0_38 */
 	status = "disabled";
 };
 
@@ -92,7 +99,6 @@
 	pinctrl-0 = <&rpi_header_ehrpwm1_pins_default>;
 };
 
-
 &main_gpio0 {
 	status = "okay";
 	pinctrl-names = "default";
-- 
2.39.2

